File: loadercomplex-fabric/ver17/src/main/java/com/enderzombi102/loadercomplex/fabric17/impl/FabricServer.java
package com.enderzombi102.loadercomplex.fabric17.impl;

import com.enderzombi102.loadercomplex.api.utils.Server;
import net.minecraft.server.MinecraftServer;

public class FabricServer implements Server {
	private final MinecraftServer backingServer;

	public FabricServer(MinecraftServer server) {
		this.backingServer = server;
	}

	@Override
	public Object getObject() {
		return this.backingServer;
	}
}

File: loadercomplex-fabric/src/ver171/kotlin/com/enderzombi102/loadercomplex/fabric171/impl/server/FabricServer.kt
package com.enderzombi102.loadercomplex.fabric171.impl.server

import com.enderzombi102.loadercomplex.api.minecraft.server.Server
import net.minecraft.server.MinecraftServer

class FabricServer( private val backingServer: MinecraftServer ) : Server {
	override fun getObject(): Any =
		this.backingServer
}
